app/rdf/query-cube-preview.ts

```diff
@@ -0,0 +1,278 @@
+PREFIX sh: <http://www.w3.org/ns/shacl#>
+PREFIX time: <http://www.w3.org/2006/time#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+    
+CONSTRUCT {
+  ?cube schema:version ?cubeVersion .
+  ?cube cube:observationSet ?observationSet .
+  ?observationSet cube:observation ?observation .
+
+  ?dimension sh:path ?dimensionIri .
+  ?dimension sh:datatype ?dimensionDataType .
+  ?dimension rdf:type ?dimensionType .
+  ?dimension qudt:scaleType ?dimensionScaleType .
+  ?dimension qudt:unit ?dimensionUnit .
+  ?dimensionUnit schema:name ?dimensionUnitLabel .
+  ?dimensionUnit qudt:CurrencyUnit ?dimensionUnitIsCurrency .
+  ?dimensionUnit qudt:currencyExponent ?dimensionUnitCurrencyExponent .
+  ?dimension sh:order ?dimensionOrder .
+  ?dimension meta:dataKind ?dimensionDataKind .
+  ?dimensionDataKind rdf:type ?dimensionDataKindType .
+  ?dimensionDataKind time:unitType ?dimensionTimeUnitType .
+  ?dimension schema:name ?dimensionLabel .
+  ?dimension schema:description ?dimensionDescription .
+
+  ?observation ?observationPredicate ?observationValue .
+  ?observation ?observationPredicate ?observationValueLabel .
+} WHERE {
+  VALUES ?cube { <${iri}> }
+  FILTER(EXISTS { ?cube a cube:Cube . }) {}
+  UNION {
+    ?cube cube:observationConstraint/sh:property ?dimension .
+    ?dimension sh:path ?dimensionIri .
+    OPTIONAL { ?dimension sh:datatype ?dimensionDataType . }
+    OPTIONAL { ?dimension rdf:type ?dimensionType . }
+    OPTIONAL { ?dimension qudt:scaleType ?dimensionScaleType . }
+    OPTIONAL {
+      { ?dimension qudt:unit ?dimensionUnit . }
+      UNION { ?dimension qudt:hasUnit ?dimensionUnit . }
+      OPTIONAL { ?dimensionUnit rdfs:label ?dimensionUnitRdfsLabel . }
+      OPTIONAL { ?dimensionUnit qudt:symbol ?dimensionUnitSymbol . }
+      OPTIONAL { ?dimensionUnit qudt:ucumCode ?dimensionUnitUcumCode . }
+      OPTIONAL { ?dimensionUnit qudt:expression ?dimensionUnitExpression . }
+      OPTIONAL { ?dimensionUnit ?dimensionUnitIsCurrency qudt:CurrencyUnit . }
+      OPTIONAL { ?dimensionUnit qudt:currencyExponent ?dimensionUnitCurrencyExponent . }
+      BIND(STR(COALESCE(STR(?dimensionUnitSymbol), STR(?dimensionUnitUcumCode), STR(?dimensionUnitExpression), STR(?dimensionUnitRdfsLabel))) AS ?dimensionUnitLabel)
+      FILTER (LANG(?dimensionUnitRdfsLabel) = "en")
+    }
+    OPTIONAL { ?dimension sh:order ?dimensionOrder . }
+    OPTIONAL {
+      ?dimension meta:dataKind ?dimensionDataKind .
+      ?dimensionDataKind rdf:type ?dimensionDataKindType .
+    }
+    OPTIONAL {
+      ?dimension meta:dataKind ?dimensionDataKind .
+      ?dimensionDataKind time:unitType ?dimensionTimeUnitType .
+    }
+    ${buildLocalizedSubQuery("dimension", "schema:name", "dimensionLabel", {
+      locale,
+    })}
+    ${buildLocalizedSubQuery(
+      "dimension",
+      "schema:description",
+      "dimensionDescription",
+      { locale }
+    )}
+    FILTER(?dimensionIri != cube:observedBy && ?dimensionIri != rdf:type)
+  } UNION {
```

Comment:
this part of the query leads to many duplicate triples in the resultset. avoiding duplicates reduces the burden of post filtering (i.e. L253)